### ISSUE TITLE
[Question/Bug] allowableValues range is throwing a NoSuchMethodError

### ISSUE THREAD

**Issue by ahatzz11** [springfox/springfox#1692]
Swagger version - `1.5.12`
SpringFox version - `2.6.1`

I have a variable with a defined range for `allowableValues` in the `ApiModelProperty` annotation:

```
    @ApiModelProperty(
        value = "This is the description",
        example = "100.0",
        allowableValues = "range[0, infinity]")
    @JsonProperty("test_property")
    private double testProperty;
```

When I start up the app, the endpoint shows me this: 

![](https://puu.sh/uihoO/3b8a054347.png)

And here is the stacktrace:

```
"stack_trace":
"java.lang.NoSuchMethodError: io.swagger.models.properties.AbstractNumericProperty.setMaximum(Ljava/lang/Double;)V
at springfox.documentation.swagger2.mappers.EnumMapper.maybeAddAllowableValues(EnumMapper.java:104)
```

I found the `setMaximum` method [here in swagger-core](https://github.com/swagger-api/swagger-core/blob/master/modules/swagger-models/src/main/java/io/swagger/models/properties/AbstractNumericProperty.java#L46-L48), but it only takes in `BigDecimal`, not a `Double`, which it is currently seeing.

I also found the `safeBigDecimal()` method, which should fix any type issues that come up - [here](https://github.com/springfox/springfox/blob/master/springfox-swagger2/src/main/java/springfox/documentation/swagger2/mappers/EnumMapper.java#L66-L75). 

It seems that the maximum (in this case `infinity`) isn't properly being converted to a BigDemical for the maximum. I have also tested this with other ranges, such as `"range[0, 1]"`.

It's weird because `setMinimum` seems to work just fine, but it has an issue once it gets to `setMaximum` on line 61 in the [EnumMapper](https://github.com/springfox/springfox/blob/master/springfox-swagger2/src/main/java/springfox/documentation/swagger2/mappers/EnumMapper.java#L61). 

Hopefully an easy fix, or something I'm doing wrong that I'm not noticing!

Thanks,
Hatz

**Comment by dilipkrish** [comment]
Thats a breaking change in the swagger-core api from 1.5.10 -> 1.5.12. You may want to switch that dial back until we ship 2.7.0

**Comment by ahatzz11** [comment]
Good to know! Two things:

1. Do you know what exactly the breaking change is? 
2. I see that the `2.7.0` release is due on March 1st, 2017 - is that release still on track? If not that's fine, but I'm curious :) 

Thanks for the help again!

**Comment by dilipkrish** [comment]
> I see that the 2.7.0 release is due on March 1st, 2017 - is that release still on track? If not that's fine, but I'm curious :)

Im trying to get it out soon. There is just a ton of stuff to do. I'd atleast like to fix a majority of issues before 2.7.0 because Im planning on sunsetting support for JDK 6.

**Comment by dilipkrish** [comment]
The breaking change is that `AbstractNumericProperty.setMaximum(Double)` => `AbstractNumericProperty.setMaximum(BigDecimal)` without keeping existing api deprecated.

**Comment by ahatzz11** [comment]
Awesome, great! ✌️

To clarify something - once this all gets sorted out, `"range[0, infinity]"` will end up adding the `maximum` and `minimum` attributes on the parameters? That's currently the only warning I'm getting on my validator.

**Comment by dilipkrish** [comment]
It should would be great if you could give 2.7.0-SNAPSHOT a whirl and report back

**Comment by ahatzz11** [comment]
I was able to build `2.7.0` locally, but I'm unsure how I can get it installed so I can use it as a dependency. I'm used to doing `gradle clean install` on my projects, but... 

> Task 'install' not found in root project 'springfox'.

Could you point me in the right direction?

**Comment by dilipkrish** [comment]
You don't need to build locally... its available [in an OSS Snapshot](http://springfox.github.io/springfox/docs/snapshot/#dependencies). Gradle equivalent of `mvn clean install` => `./gradlew clean build` (again in the docs as well)

**Comment by ahatzz11** [comment]
Alright I have it working now. Thanks for the help. Observations:

1. I had a customized endpoint [like so](https://springfox.github.io/springfox/docs/snapshot/#customizing-the-swagger-endpoints) and that broke when I updated to `2.7.0-SNAPSHOT`. I ended up having to use `/v2/api-docs` to get it to work.

2. My variables have allowable values of 'only positive values'. I used `"range[0, infinity]"`. It adds in the minimum just fine, but it doesn't add the maximum. I assume because there isn't anything actually specified and the limitation would then be based on the type (int, long, etc). Is this intended behavior? My validator still tells me I don't have a maximum, so I might have to change `infinity` to an actual value, or just live with the validator warning.

3. I submitted #1651 a while back, and that seems to be resolved! Thanks for the fix 👍
